src/main/java/com/epam/test_generator/dao/impl/JiraStoryDAO.java

```diff
@@ -0,0 +1,135 @@
+package com.epam.test_generator.dao.impl;
+
+import com.epam.test_generator.dao.interfaces.SuitDAO;
+import com.epam.test_generator.entities.Suit;
+import com.epam.test_generator.pojo.JiraStory;
+import net.rcarz.jiraclient.*;
+import net.rcarz.jiraclient.Issue.SearchResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.MalformedParametersException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class JiraStoryDAO {
+
+    @Autowired
+    private SuitDAO suitDAO;
+
+    @Autowired
+    private JiraClient client;
+
+
+    private final static String TYPE = "Story";
+    private final static String uri = "https://jirapct.epam.com/jira";
+
+
+    public JiraStory getStoryByJiraKey(String jiraKey) {
+        try {
+            return new JiraStory(client.getIssue(jiraKey));
+        } catch (JiraException e) {
+
+            if (e.getCause() instanceof RestException) {
+                RestException restException = (RestException) e.getCause();
+                if (restException.getHttpStatusCode() == 404) return null;
+            }
+            throw new MalformedParametersException(e.getMessage());
+        }
+    }
+
+    public List<JiraStory> getStories(String jiraProjectKey) {
+        List<Issue> issues;
+        try {
+            issues = client.searchIssues("project=" + jiraProjectKey +" AND status in (Open, \"In Progress\", Reopened, Verified) AND type=story ", 1000).issues;
+        } catch (JiraException e) {
+            throw new MalformedParametersException(e.getMessage());
+        }
+        List<JiraStory> jiraStories = new ArrayList<>();
+        issues.forEach(s -> jiraStories.add(new JiraStory(s)));
+        return jiraStories;
+    }
+
+    public List<JiraStory> getUnexistedStoriesByProject(String projectKey) {
+        List<JiraStory> jiraStories = new ArrayList<>();
+        try {
+            SearchResult issues = client.searchIssues("project = " + projectKey + " AND status in (Open, \"In Progress\", Reopened, Verified) AND type=story ", 1000);
+            issues.issues.forEach(story -> {
```

Comment:
This would be correct usage of stream api:

    issues.issues.stream()
        .filter(story -> suitDAO.findByJiraKey(story.getKey()) = null)
        .map(story -> new JiraStory(story))
        .collect(Collectors.toList());

If you not sure just use pre-lambda style.